resolwe_bio/processes/samtools/samtools_depth.py

```diff
@@ -18,6 +21,13 @@
 )
 
 
+def prune_zero_depth(stdout):
+    """Prune zero depth entries from the samtools depth output."""
+    df = pd.read_csv(StringIO(stdout), sep="\t", names=["chrom", "pos", "depth"])
```

Comment:
To avoid reading the entire file, you can limit the import to non-zero values, perhaps something along the lines of (untested):

```python
import pandas as pd

chunkz = pd.read_csv(StringIO(stdout), sep="\t", names=["chrom", "pos", "depth"], chunksize=5000)
df = pd.concat((x.query("depth > 0") for x in chunkz))
```

This works, though:

```python
from io import StringIO
import pandas as pd


file_name = "https://raw.githubusercontent.com/uiuc-cse/data-fa14/gh-pages/data/iris.csv"
df = pd.read_csv(file_name)

flow = StringIO()
xy = df.to_csv(flow)

chunkz = pd.read_csv(flow , chunksize=5)
pd.concat((x.query("species == 'setosa'") for x in chunkz))

Unnamed: 0	sepal_length	sepal_width	petal_length	petal_width	species
0	0	5.1	3.5	1.4	0.2	setosa
1	1	4.9	3.0	1.4	0.2	setosa
2	2	4.7	3.2	1.3	0.2	setosa
3	3	4.6	3.1	1.5	0.2	setosa
4	4	5.0	3.6	1.4	0.2	setosa
5	5	5.4	3.9	1.7	0.4	setosa
6	6	4.6	3.4	1.4	0.3	setosa
...
```